Libs/MRML/Core/Testing/vtkMRMLNodeTest1.cxx

```diff
@@ -2854,6 +2863,69 @@ bool TestClearScene()
   return true;
 }
 
+//----------------------------------------------------------------------------
+int TestReadWriteXMLProperties()
+{
+  vtkNew<vtkMRMLNodeTestHelper1> node1;
+
+  std::vector<std::string> testingStringsVector;
+  testingStringsVector.push_back("Several;Semicolon;In;One;String");
+  testingStringsVector.push_back("Percent%Sign");
+  testingStringsVector.push_back("%25%3B;%");
+  node1->SetTestingStringVector(testingStringsVector);
+
+  std::vector<float> testingFloatVector;
+  testingFloatVector.push_back(0.0);
+  testingFloatVector.push_back(1.1);
+  testingFloatVector.push_back(2.2);
+  testingFloatVector.push_back(3.3);
+  testingFloatVector.push_back(4.4);
+  testingFloatVector.push_back(5.5);
+  node1->SetTestingFloatVector(testingFloatVector);
+
+  std::vector<int> testingIntVector;
+  testingFloatVector.push_back(0);
+  testingFloatVector.push_back(1);
+  testingFloatVector.push_back(2);
+  testingFloatVector.push_back(3);
+  testingFloatVector.push_back(4);
+  testingFloatVector.push_back(5);
+  node1->SetTestingIntVector(testingIntVector);
+
+  std::stringstream ss;
+  ss << "<MRMLNode ";
+  node1->WriteXML(ss, NULL);
+  ss << " />";
+  std::string testString = ss.str();
+  vtkSmartPointer<vtkXMLDataElement> element = vtkSmartPointer<vtkXMLDataElement>::Take(vtkXMLUtilities::ReadElementFromStream(ss));
+
+  const char *atts[] = {
+    "TestingStringVector", element->GetAttribute("TestingStringVector"),
+    "TestingFloatVector", element->GetAttribute("TestingFloatVector"),
+    "TestingIntVector", element->GetAttribute("TestingIntVector"),
+    NULL };
+
+  vtkNew<vtkMRMLNodeTestHelper1> node2;
+  node2->ReadXMLAttributes(atts);
```

Comment:
It seems `atts` is not used anywhere

Reply:
atts is used in:
```node2->ReadXMLAttributes(atts);```
To populate the parameters of node2.